### PR TITLE
Fixes #12257 - added Debian kexec template

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

**Comment by stbenjam** [comment]
I get:

```
2015-10-22T10:41:25 [app] [W] Reloading kernel on mac5254002b5b69.example.com task failed with the following error: un
defined method `url_for_boot' for #<Debian:0x000000023cd618>
 | NoMethodError: undefined method `url_for_boot' for #<Debian:0x000000023cd618>
```

For whatever reason Debian OS's don't have this url_for_boot method? Everyone else seems to

**Comment by lzap** [comment]
Sorry this took so long :-) @mmoll actually pointed out on the list how this is solved in iPXE template, I copied the trick over here:

```
boot_files_uris = @host.operatingsystem.boot_files_uri(@host.medium, @host.architecture)
```

https://github.com/theforeman/community-templates/blob/develop/preseed/iPXE.erb

Rebased, hopefully it works now.

File: app/views/foreman_discovery/debian_kexec.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<%#
+kind: kexec
+name: Debian kexec
+oses:
+oses:
```
Comment (mmoll) [inline app/views/foreman_discovery/debian_kexec.erb]: duplicate line

File: app/views/foreman_discovery/debian_kexec.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Debian 6.0
+- Debian 7.
```
Comment (mmoll) [inline app/views/foreman_discovery/debian_kexec.erb]: in the community-templates, we did cut that down to just Debian and Ubuntu to let all versions associate

**Comment by lzap** [comment]
Develop branch is utterly broken right now, need to take a look.

**Comment by lzap** [comment]
This can be now tested [test]

**Comment by lzap** [comment]
Please ignore the failure, its javascript error.

**Comment by dLobatog** [comment]
I'm going to test with Debian & Ubuntu & report now..

**Comment by dLobatog** [comment]
Holding off because I found http://projects.theforeman.org/issues/15644 trying this out

**Comment by lzap** [comment]
What's the status? Have you provided correct link above? I think this one is not related.

**Comment by dLobatog** [comment]
@lzap Yup, sorry. I'll try this one tomorrow. The ticket I linked to wasn't caused by this, but I couldn't keep on testing without the fix because the discovery host facts couldn't be submitted without the patch.

**Comment by lzap** [comment]
Fixed finally, the problem was `url_for_boot` was not available for Debian. Also, both @kernel and @initrd variables are overriden when doing template preview, this was very confusing. It is only set properly during provisioning of a discovered host. Therefore I changed the templates to set the fields properly themself, I am keeping the two attributes just for compatibility.

During testing I found two regressions, you need to merge them before testing this. I will link them both to this PR.

**Comment by lzap** [comment]
Test this with Katello installed if possible as it overrides some mediapath etc.

**Comment by lzap** [comment]
@dLobatog @stbenjam since this one is quite work to test, I advise you to review and merge the other three PRs with Important label and do this as the last one if possible. I can perhaps make a Discovery 6.0 release without this patch, but the others are more important. Thanks!

**Comment by lzap** [comment]
Rebased, thanks for reviews @dLobatog !

**Comment by dLobatog** [comment]
Success - please add to the release notes that it also works with Ubuntu (I tried with 14.04 & it worked). Great job, merging!

**Comment by dLobatog** [comment]
Merged as e940d32c6e2958e3ab048b1d5c13689748d4dc9e, thanks @lzap!

**Pull request closed without merging**